'make env' fails with latest core for esp32

```
esp32:esp32@1.0.5 installed
cp: cannot create regular file '/home/runner/work/msp-firmware/msp-firmware/var/packages/esp32/hardware/esp32/1.0.4/': Not a directory
make: *** [Makefile:119: env] Error 1
```

The error comes from the hardcoded path in `Makefile` where it automates the workaround described by [BSEC-Arduino-Library](https://github.com/BoschSensortec/BSEC-Arduino-library#3-modify-the-platformtxt-file) documentation.

According to [ESP32 core 1.0.5 release notes](https://github.com/espressif/arduino-esp32/releases/tag/1.0.5), the above issue should have been fixed in commit https://github.com/espressif/arduino-esp32/commit/9e65ed1a, but just dropping the workaround from `Makefile` does not seem to be enough. Need further investigation.

Options to fix:
* stick with ESP32 core 1.04 (pin dependency version)
* keep the workaround and make it core version-agnostic (by figuring out the correct destination path for the `platform.local.txt` file)
* hopefully drop the workaround at all...

--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,8 @@ LIBRARIES_URLS := \
 	https://github.com/A-A-Milano-Smart-Park/MiCS6814-I2C-MOD-Library
 
 # The FQBN is the core, followed by the board.
-FQBN := $(CORE):$(BOARD)
+CORE_NAME := $(shell echo $(CORE) | cut -f1 -d@)
+FQBN := $(CORE_NAME):$(BOARD)
 
 # Treat all warnings as errors.
 BUILDPROP := compiler.warning_flags.all='-Wall -Wextra'

--- a/Makefile
+++ b/Makefile
@@ -10,7 +10,7 @@ PORT := /dev/ttyACM0
 ADDITIONAL_URLS := \
 	https://dl.espressif.com/dl/package_esp32_index.json \
 	http://arduino.esp8266.com/stable/package_esp8266com_index.json
-CORES := esp32:esp32
+CORE := esp32:esp32
 LIBRARIES := \
 	"BSEC Software Library" \
 	"PMS Library" \
@@ -18,8 +18,8 @@ LIBRARIES := \
 LIBRARIES_URLS := \
 	https://github.com/A-A-Milano-Smart-Park/MiCS6814-I2C-MOD-Library
 
-# The FQBN is the first specified core, followed by the board.
-FQBN := $(word 1, $(CORES)):$(BOARD)
+# The FQBN is the core, followed by the board.
+FQBN := $(CORE):$(BOARD)
 
 # Treat all warnings as errors.
 BUILDPROP := compiler.warning_flags.all='-Wall -Wextra'
@@ -112,7 +112,7 @@ endif
 env: $(BINDIR)/arduino-cli $(BINDIR)/arduino-lint $(ETCDIR)/arduino-cli.yaml
 	mkdir -p $(BUILDDIR)
 	$(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml core update-index
-	$(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml core install $(CORES)
+	$(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml core install $(CORE)
 	# add platform.local.txt to installed esp32 platform
 	# required by BSEC Arduino Library
 	# https://github.com/BoschSensortec/BSEC-Arduino-library#3-modify-the-platformtxt-file

--- a/Makefile
+++ b/Makefile
@@ -117,7 +117,8 @@ env: $(BINDIR)/arduino-cli $(BINDIR)/arduino-lint $(ETCDIR)/arduino-cli.yaml
 	# add platform.local.txt to installed esp32 platform
 	# required by BSEC Arduino Library
 	# https://github.com/BoschSensortec/BSEC-Arduino-library#3-modify-the-platformtxt-file
-	cp platform.local.txt $(VARDIR)/packages/esp32/hardware/esp32/1.0.4/
+	CORE_VERSION=$$($(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml core list | grep "$(CORE_NAME)" | cut -f2 -d' ') ; \
+	cp platform.local.txt $(VARDIR)/packages/esp32/hardware/esp32/$$CORE_VERSION/
 ifdef LIBRARIES
 	$(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml lib update-index
 	$(BINDIR)/arduino-cli --config-file $(ETCDIR)/arduino-cli.yaml lib install $(LIBRARIES)

